src/capi/TuriCore.h

```diff
@@ -33,8 +36,8 @@ struct tc_flex_dict_struct;
 typedef struct tc_flex_dict_struct tc_flex_dict;
 
 // datetime
-struct tc_datetime_struct;
-typedef struct tc_datetime_struct tc_datetime;
+struct tc_datetime { struct tc_impl_base* impl; };
```

Comment:
This value a polymorphic base type with a variable pointer size between C and C++.  Something like this would work, but we need a bit more indirection, I think.